### DISCUSSION TITLE
Does BlackSheep adhere to Semantic Versioning?

### DISCUSSION THREAD

**Discussion by netanel-haber** [Neoteroi/BlackSheep#478] [General]
I recently unpinned BlackSheep from `==2.0.4` to `~=2.0.4`, and `2.0.6` was installed, which includes the default signal handler registry. This broke my testing flow, where I start blacksheep servers in a `Thread` (maybe a bad idea, but that's another discussion I guess):

> [Besides, only the main thread of the main interpreter is allowed to set a new signal handler.](https://docs.python.org/3/library/signal.html#signals-and-threads)

I was wondering what versioning scheme BlackSheep adheres to? I couldn't find any mention in the docs. I think the additions in `2.0.6` should at least be considered features by the standard of semver.

Thanks!

**Comment by RobertoPrevato** [comment]
Replied under https://github.com/Neoteroi/BlackSheep/issues/481
